api/src/main/java/net/kyori/adventure/text/JoinConfigurationImpl.java

```diff
@@ -153,13 +164,21 @@ public String toString() {
     int componentsSeen = 0;
 
     if (!it.hasNext()) {
-      return singleElementJoin(config, component);
+      return singleElementJoin(
+        config,
+        hasRootStyle ?
```

Comment:
I think we should handle this in the `singleElementJoin` method. Otherwise, the root style won't apply to the prefix/suffix/etc (which is the case for multi-element joining).